truss/templates/control/control/application.py

```diff
@@ -43,53 +45,61 @@ async def handle_model_load_failed(_, error):
 
 
 def create_app(base_config: Dict):
-    app = FastAPI(
-        title="Truss Live Reload Server",
-        exception_handlers={
-            PatchApplicatonError: handle_patch_error,
-            ModelLoadFailed: handle_model_load_failed,
-            Exception: generic_error_handler,
-        },
-    )
+    app_state = State()
 
     setup_logging()
 
     app_logger = logging.getLogger(__name__)
 
-    app.state.logger = app_logger
+    app_state.logger = app_logger
 
     for k, v in base_config.items():
-        setattr(app.state, k, v)
+        setattr(app_state, k, v)
 
-    app.state.inference_server_process_controller = InferenceServerProcessController(
-        app.state.inference_server_home,
-        app.state.inference_server_process_args,
-        app.state.inference_server_port,
+    app_state.inference_server_process_controller = InferenceServerProcessController(
+        app_state.inference_server_home,
+        app_state.inference_server_process_args,
+        app_state.inference_server_port,
         app_logger=app_logger,
     )
 
     limits = httpx.Limits(max_keepalive_connections=8, max_connections=32)
 
-    app.state.proxy_client = httpx.AsyncClient(
-        base_url=f"http://localhost:{app.state.inference_server_port}", limits=limits
+    app_state.proxy_client = httpx.AsyncClient(
+        base_url=f"http://localhost:{app_state.inference_server_port}", limits=limits
     )
 
-    pip_path = getattr(app.state, "pip_path", None)
+    pip_path = getattr(app_state, "pip_path", None)
 
     patch_applier = ModelContainerPatchApplier(
-        Path(app.state.inference_server_home),
+        Path(app_state.inference_server_home),
         app_logger,
         pip_path,
     )
 
-    oversee_inference_server = getattr(app.state, "oversee_inference_server", True)
+    oversee_inference_server = getattr(app_state, "oversee_inference_server", True)
 
-    app.state.inference_server_controller = InferenceServerController(
-        app.state.inference_server_process_controller,
+    app_state.inference_server_controller = InferenceServerController(
+        app_state.inference_server_process_controller,
         patch_applier,
         app_logger,
         oversee_inference_server,
     )
+
+    def startup_routine():
+        inference_server_startup_flow(app_state.inference_server_controller, app_logger)
+
+    app = FastAPI(
+        title="Truss Live Reload Server",
+        on_startup=[startup_routine],
```

Comment:
inference_server_startup_flow pings the external service to provide patches, so it's desirable that control server is up by then. So we should make sure that on_startup execution happens in the background and doesn't block the control server startup, otherwise caller will have to unnecessarily go through timeouts or failures. I think it should still work because the caller should retry but would be good to avoid. Previously this was done in background using a separate thread. Would be good to check how FastApi executed this function. If that's happening on the executor (thread pool used for blocking tasks) then that's likely fine.